### PR TITLE
Add support for custom inline delimiters and raw arguments

### PR DESCRIPTION
**Pull request by mbovel**
Less code, more features, faster than `JSON.parse()` 🎉

### CODE REVIEW

File: src/InlineParser.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+interface InlineSyntax {
+	separator?: string;
+	end: string;
+}
+
+export class InlineParser {
+	private readonly regex: RegExp;
+	private readonly inlineTag: InlineSyntax = { separator: "][", end: "]" };
+	private readonly sugars: Map<string, Sugar>;
+	private stack: Array<{ element: InlineElement; syntax: InlineSyntax }> = [];
```
Comment (MaximeKjaer) [inline src/InlineParser.ts]: I would rename this to `openInlinesStack`. Or perhaps we could define a `type Stack<T> = Array<T>` to syntactically show that it's a stack?

File: src/InlineParser.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+						this.close(j + 1);
+						this.openArg();
+						return;
+					}
+				}
+
+				const sugar = this.sugars.get(token);
+				if (sugar && this.isSugarStart(sugar)) {
+					this.open(sugar.tag, sugar);
+					return;
```
Comment (MaximeKjaer) [inline src/InlineParser.ts]: I would prefer `if (cond) {x} else {y}` rather than `if (cond) { x; return} y` here

File: test/parseTest.ts [hunk 3]
```diff
@@ -1,12 +1,63 @@
...
+import { defaultSchema } from "../src/schema";
 import { FileInfo, filesIn } from "./utils";
 
-describe("parse()", () => {
+/* tslint:disable:only-arrow-functions */
+
+describe("parse()", function() {
 	for (const output of filesIn("test/output/parse", ".json")) {
-		it(`works with ${output.name}`, () => {
+		beforeEach(function() {
```
Comment (MaximeKjaer) [inline test/parseTest.ts]: This should be `before`, not `beforeEach`

File: src/InlineParser.ts (hunk 1 above)
Comment (MaximeKjaer) [reply src/InlineParser.ts]: Actually, I'm fine with this. The comment I added is enough context.

File: test/parseTest.ts (hunk 3 above)
Comment (mbovel) [reply test/parseTest.ts]: Fixed 😄

**Pull request merged**